include/deal.II/lac/trilinos_tpetra_vector.templates.h

```diff
@@ -411,6 +411,31 @@ namespace LinearAlgebra
 
 
 
+    template <typename Number, typename MemorySpace>
+    template <typename OtherNumber>
+    Vector<Number, MemorySpace> &
+    Vector<Number, MemorySpace>::operator=(const dealii::Vector<OtherNumber> &V)
+    {
+#  ifdef DEBUG
+      bool number_type_matches = std::is_same_v<OtherNumber, Number>;
+      Assert(number_type_matches, ExcInternalError());
+#  endif
```

Comment:
This function wouldn't compile anyway if the number typoes don't match because of
```
vector = Utilities::Trilinos::internal::make_rcp<VectorType>(
        V.locally_owned_elements().make_tpetra_map_rcp(), vector_data);
```
right?

Reply:
I was not sure. Therefore, I added that assertion.

But you are right now when I think about it, the function 
```
vector = Utilities::Trilinos::internal::make_rcp<VectorType>(
        V.locally_owned_elements().make_tpetra_map_rcp(), vector_data);
```
should lead to a linking error, when the number type does not match.
So we can remove those asserts.